### DISCUSSION TITLE
Use models after training to classify unknown data

### DISCUSSION THREAD

**Discussion by padmalcom** [safe-graph/GNN-FakeNews#3] [Q&A]
Hi, first of all: Thanks for your work! I'm new to GNNs in the domain of fake new detection and I could reproduce the training. My question is now what to do next with those models? Or how can I check a news article if it is fake or not?

**Comment by YingtongDou** [comment]
Hi, thanks for your interest in our project!

There are two future directions you can work on given our repo:

**1. Methodology-oriented:**
Given the UPFD dataset, you can split the existing data into training&test sets and devise a better GNN model that can hit higher performance on the test set.

**2. Application-oriented:**
Following the same approaches described in our UPFD [paper](https://arxiv.org/pdf/2104.12259.pdf), you can crawl news posts and their engaged tweets on Twitter and construct news propagation graphs. Then you could train and test the GNN models introduced in this repo on a new set of news.

**Comment by padmalcom** [comment]
@YingtongDou thank you for the fast reply. I'd go with the application oriented approach. Can you tell me how I can:
- construct a new propagation graph?
- test the GNN models from this repo on a new set of news?
Or provide any sources that adress these tasks?

I got so far that I collected data from twitter, created a custom data object using bag of words as feature but I'm struggling how to apply the model I train to new data. Normally, I'd to a _fit_ and _transform_ in machine learning. But for GNNs I only find information on how to do the _fit_, not how to do the _transform_.

**Comment by YingtongDou** [comment]
- construct a new propagation graph?
Please refer to Section 2.2 in [paper](https://arxiv.org/pdf/2104.12259.pdf) for details on constructing a news propagation graph. 
Moreover, you need to check out the [data_loader.py](https://github.com/safe-graph/GNN-FakeNews/blob/main/utils/data_loader.py) to figure out how to transform the collected data (adj matrix, node features, news labels) into pyg data format.

- test the GNN models from this repo on a new set of news?
Testing on a new set of news also starts from collecting news data from Twitter and then building the news propagation graph. During training the model on old data, you can save the variables of the best-performed model. Then you can load the best-performed model to classify the newly collected news based on their propagation graphs.

**Reply by padmalcom** [reply]
Thank you @YingtongDou
